### ISSUE TITLE
Writing csv files as the code and plot for article

### ISSUE THREAD

**Issue by DorisAmoakohene** [DorisAmoakohene/Doris-Amoakohene-atime-for-asymptotic-Timings#9]
@tdhock see this and let me know what you think about using this as the figure and 2

![refs write vary cols R](https://github.com/user-attachments/assets/b695c789-7351-4730-a620-a4f9db15ce84)
![new gg](https://github.com/user-attachments/assets/6d751df8-c1b8-4d85-976f-a8c070a0e3f2)

This plot above are plotting for ; 
```
refs.write.vary.cols.R <- atime::references_best(atime.write.vary.cols.R)
```
which one looks good to show

![gg write R](https://github.com/user-attachments/assets/f0d7a857-f61f-4f82-b6e5-9164697b9921)

**Comment by tdhock** [comment]
I don't think it is a good example because there is no asymptotic complexity class difference.
it would be better to show an example where one curve is clearly linear, other is clearly quadratic.

**Comment by DorisAmoakohene** [comment]
very well, will investigate other packages and send another result

**Comment by tdhock** [comment]
regex TRE vs PCRE vs RE2 would be good

**Comment by DorisAmoakohene** [comment]
hi @tdhock is this what you meant?

subject.size.vec <- unique(as.integer(10^seq(0, 3.5, l=100)))

atime.list <- atime::atime(
  N=subject.size.vec, # Vector of sizes.
  setup={ # Run for each size, before timings:
    subject <- paste(rep("a", N), collapse="")
    pattern <- paste(rep(c("a?", "a"), each=N), collapse="")
  },
  times=10, 
  seconds.limit=0.1, 

  PCRE.match=regexpr(pattern, subject, perl=TRUE),
  TRE.match=regexpr(pattern, subject, perl=FALSE),
  RE2.match=re2::re2_match(subject, pattern)
)


plot(atime.list)


(best.list <- atime::references_best(atime.list))
plot(best.list)

<img width="475" alt="Screenshot 2024-09-23 121909" src="https://github.com/user-attachments/assets/a3ab40b3-8803-4296-b932-d64e42280619">

**Comment by tdhock** [comment]
yes I think that example would be good because you can clearly see the difference between PCRE (exponential time) and others (cubic)
please remove ".match" from the names
also you can probably remove RE2 to simplify (it is the same N^3 asymptotic time complexity as TRE)

**Comment by DorisAmoakohene** [comment]
1. ![best list](https://github.com/user-attachments/assets/b93a209f-38bc-4eb6-a478-a0f97992f97a)
is it okay to show it like this

or like this 
2. 
![best_fit_asymptotic_time_complexity](https://github.com/user-attachments/assets/e55c94e7-70b0-400b-b102-81ffd2ad458a)

**Comment by tdhock** [comment]
ok.
the first plot x min is 10 but would be better 1
both probably need bigger text

**Comment by DorisAmoakohene** [comment]
> ok. the first plot x min is 10 but would be better 1 both probably need bigger text

![best list](https://github.com/user-attachments/assets/722ba8c5-262a-4328-b986-357987419a39)

**Comment by tdhock** [comment]
there is no N=1 on x axis?

**Comment by DorisAmoakohene** [comment]
why do I get this error when I run this 

```
atime.list[["measurements"]][N==323, .(expr.name, seconds=median, kilobytes)]
> pred.list <- predict(best.list, seconds=1e-2, kilobytes=10)
```
```
Error in `[.data.table`(data.table(unit = names(L)), , { : 
  kilobytes is not a valid unit; argument names of predict must be one of these valid units: seconds
```

**Comment by tdhock** [comment]
your machine does not support memory measurement for some reason

**Comment by DorisAmoakohene** [comment]
@tdhock including code in the overleaf like

```
\begin{Schunk}
\begin{Sinput}

subject.size.vec <- unique(as.integer(10^seq(0, 2, l=10)))

atime.list <- atime::atime(
  N = subject.size.vec,
  setup = {
    subject <- paste(rep("a", N), collapse="")
    pattern <- paste(rep(c("a?", "a"), each=N), collapse="")
  },
  times = 10,
  seconds.limit = 0.1,
  PCRE = regexpr(pattern, subject, perl=TRUE),
  TRE = regexpr(pattern, subject, perl=FALSE)
)
\end{Sinput}
\end{Schunk}
```

is giving me errors like:

```
LaTeX Error: File Sweave.sty' not found.

LaTeX Error: Environment shunk undefined.

‪./DorisAmoakohene.tex, 230‬
You have created an environment (using \begin{…} and \end{…} commands) which is not recognized. Make sure you have included the required package for that environment in your preamble, and that the environment is spelled correctly.

```

I have spent time looking at it and find solution buts taking too much time, Can you assist

**Comment by tdhock** [comment]
you need to install Sweave.sty in your latex
on overleaf you should upload the file

**Comment by DorisAmoakohene** [comment]
\usepackage{Sweave} I have this in my in my wrapper.tex file but I still that error
